tensorflow/lite/micro/kernels/reduce_test.cc

```diff
@@ -74,20 +74,96 @@ void TestMeanFloatInput4D(const int* input_dims_data, const float* input_data,
   TfLiteIntArray* output_dims = IntArrayFromInts(output_dims_data);
   const int output_dims_count = ElementCount(*output_dims);
 
+  TfLiteRegistration registration = tflite::ops::micro::Register_MEAN();
+
+  constexpr int num_of_inputs = 2;   // input and axis
+  constexpr int num_of_outputs = 1;  // output
+
+  constexpr int tensors_size = num_of_inputs + num_of_outputs;
+  TfLiteTensor tensors[tensors_size] = {
+      CreateFloatTensor(input_data, input_dims),
+      CreateInt32Tensor(axis_data, axis_dims),
+      CreateFloatTensor(output_data, output_dims),
+  };
+
+  TF_LITE_MICRO_EXPECT_EQ(
+      kTfLiteOk, ValidateReduceGoldens(
+                     tensors, tensors_size, expected_output_data, output_data,
+                     output_dims_count, registration, params, tolerance));
+}
+
+void TestReduceOpFloat(const int* input_dims_data, const float* input_data,
+                       const int* axis_dims_data, const int32_t* axis_data,
+                       const int* output_dims_data,
+                       const float* expected_output_data,
+                       TfLiteRegistration registration,
+                       TfLiteReducerParams* params, float tolerance = 1e-5) {
+  TfLiteIntArray* input_dims = IntArrayFromInts(input_dims_data);
+  TfLiteIntArray* axis_dims = IntArrayFromInts(axis_dims_data);
+  TfLiteIntArray* output_dims = IntArrayFromInts(output_dims_data);
+  const int output_dims_count = ElementCount(*output_dims);
+
   constexpr int num_of_inputs = 2;   // input and axis
   constexpr int num_of_outputs = 1;  // output
 
+  float output_data[output_dims_count];
```

Comment:
pass this in as a parameter to this function so that the size is known at compile time (which would be the case if the array were defined in the test case). This might work with some compilers but fail with others -- we have had this issue in the past in other parts of the codebase.